WebHostLib/templates/playerOptions/macros.html

```diff
@@ -0,0 +1,264 @@
+                    <div class="option-divider">&nbsp;</div>
+                {% endif %}
+                {% for location_name in world.location_names|sort %}
+                    <div class="option-entry">
+                        <input type="checkbox" id="{{ option_name }}-{{ location_name }}" name="{{ option_name }}" value="{{ location_name }}" {{ "checked" if location_name in option.default }} />
+                        <label for="{{ option_name }}-{{ location_name }}">{{ location_name }}</label>
+                    </div>
+                {% endfor %}
+            </div>
+        </td>
+    </tr>
+{% endmacro %}
+
+{% macro ItemSet(option_name, option, world) %}
+    <tr>
+        <td>{{ OptionTitle(option_name, option) }}</td>
+        <td>
+            <div class="option-container">
+                {% for group_name in world.item_name_groups.keys()|sort %}
+                    {% if group_name != "Everything" %}
+                        <div class="option-entry">
+                            <input type="checkbox" id="{{ option_name }}-{{ group_name }}" name="{{ option_name }}" value="{{ group_name }}" {{ "checked" if group_name in option.default }} />
+                            <label for="{{ option_name }}-{{ group_name }}">{{ group_name }}</label>
+                        </div>
+                    {% endif %}
+                {% endfor %}
+                {% if world.item_name_groups.keys()|length > 1 %}
+                    <div class="option-divider">&nbsp;</div>
+                {% endif %}
+                {% for item_name in world.item_names|sort %}
+                    <div class="option-entry">
+                        <input type="checkbox" id="{{ option_name }}-{{ item_name }}" name="{{ option_name }}" value="{{ item_name }}" {{ "checked" if item_name in option.default }} />
+                        <label for="{{ option_name }}-{{ item_name }}">{{ item_name }}</label>
+                    </div>
+                {% endfor %}
+            </div>
+        </td>
+    </tr>
+{% endmacro %}
+
+{% macro OptionSet(option_name, option) %}
+    <tr>
+        <td>{{ OptionTitle(option_name, option) }}</td>
+        <td>
+            <div class="option-container">
+                {% for key in option.valid_keys %}
+                    <div class="option-entry">
+                        <input type="checkbox" id="{{ option_name }}-{{ key }}" name="{{ option_name }}" value="{{ key }}" {{ "checked" if key in option.default }} />
+                        <label for="{{ option_name }}-{{ key }}">{{ key }}</label>
+                    </div>
+                {% endfor %}
+            </div>
+        </td>
+    </tr>
+{% endmacro %}
+
+{% macro OptionTitle(option_name, option) %}
+    <label for="{{ option_name }}">
+        {{ option.display_name|default(option_name) }}:
+        <span class="interactive" data-tooltip="{{ (option.__doc__ | default("Please document me!"))|escape }}">(?)</span>
```

Comment:
```suggestion
        <span class="interactive" data-tooltip="{{ (option.__doc__.strip().replace('    ', '') | default("Please document me!"))|escape }}">(?)</span>
```
Optional, I believe this should replicate the tooltip formatting of current. Not really a fan of the `replace` though. Probably needs an equivalent one for weighted options.

Reply:
I think `dedent` is what we want.

https://docs.python.org/3/library/textwrap.html#textwrap.dedent

Reply:
Added the `dedent` filter. Weighted options actually doesn't use tooltips, so it isn't needed there.